sdk/core/Azure.Core/tests/OperationTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core.Testing;
+using Azure.Core.Tests.TestFramework;
+using NUnit;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class OperationTests
```

Comment:
It'd be good to add a test that verifies you can't get the `Value` before `HasValue` is `true` as well

Reply:
will do. having said that these tests test more the operation subclass than this abstraction. I still think it's worth having the tests, so that actually implementations know what the contract looks like. I was even thinking about writing a contract verification reusable test, but I would like to have a couple of real implementations before I do that.